aws_common/include/aws_common/sdk_utils/parameter_reader.h

```diff
@@ -20,6 +20,114 @@
+     *   ParameterPath("config", "timeout")
+     * @tparam Args
+     * @param parameter_path_keys
+     */
+    template<typename ...Args>
+    ParameterPath(Args... parameter_path_keys) :
+      parameter_path_keys_(std::vector<std::string>{(parameter_path_keys)...}) {}
+
+    /**
+     * Returns the parameter's whereabouts in the node namespace hierarchy.
+     * @note Only applies if node_namespaces was specified during construction; otherwise, an empty string is returned.
+     * @param node_namespace_separator
+     * @return
+     */
+    std::string get_node_path(char node_namespace_separator) const
+    {
+        std::string resolved_path;
+        /* Construct the node's path by the provided lists of keys */
+        for (auto key = node_namespaces_.begin(); key != node_namespaces_.end(); key++) {
+            resolved_path += *key + node_namespace_separator;
+        }
+        if (!resolved_path.empty() && resolved_path.back() == node_namespace_separator) {
+            resolved_path.pop_back();
+        }
+        return resolved_path;
+    }
+    /**
+     * Return the parameter path including parameter namespaces but excluding node's namespaces.
+     * @note Only applies if parameter_path_keys was specified during construction; otherwise, an empty string is returned.
+     * @param parameter_namespace_separator
+     * @return
+     */
+    std::string get_local_path(char parameter_namespace_separator) const
+    {
+        std::string resolved_path;
+        /* Construct the parameter's path by the provided lists of keys */
+        for (auto key = parameter_path_keys_.begin(); key != parameter_path_keys_.end(); key++) {
+            resolved_path += *key + parameter_namespace_separator;
+        }
+        if (!resolved_path.empty() && resolved_path.back() == parameter_namespace_separator) {
+            resolved_path.pop_back();
+        }
+        return resolved_path;
+    }
+    /**
+     * Resolves & returns the parameter's path. Supports two separate use cases:
+     *  1. A single, 'flat' string was provided upon construction.
+     *    In this case, we return it as is, while converting node namespaces to parameter namespaces if needed.
+     *  2. Detailed lists of strings were specified for the parameter & node namespaces.
+     *    In this case, we construct the resolved path using the provided separators. Conversion would not apply.
+     * @param node_namespace_separator
+     * @param parameter_namespace_separator
+     * @param convert_node_ns_to_parameter_ns Only applies if the path was specified as a single string,
+     *   and the string contains node separators but does not contain any parameter separators.
+     * @return string representing the full, resolved path of the parameter.
+     * @note If node_namespaces and parameter_path_keys are empty, an empty string would be returned.
+     */
+    std::string get_resolved_path(char node_namespace_separator,
```

Comment:
Is this intended to be able to convert ROS2->ROS1 paths as well? If not we should specify that in the constructor  and deprecate it. This get_resolved_path almost is able to do both, and could be abused that way.
Also why don't we just put get_resolved_path within the rosparamreader itself, that way the param reader can choose how to format the parameter path?